File: src/main/java/com/senderman/Command.java
package com.senderman;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
public @interface Command {
    String name();
    String desc();
    boolean showInHelp() default true;
    boolean forAllAdmins() default false;
    boolean forPremium() default false;
    boolean forMainAdmin() default false;
}

File: src/main/java/com/senderman/Command.kt
package com.senderman

@Target(AnnotationTarget.FUNCTION)
@kotlin.annotation.Retention(AnnotationRetention.RUNTIME)
annotation class Command(
        val name: String,
        val desc: String,
        val showInHelp: Boolean = true,
        val forAllAdmins: Boolean = false,
        val forPremium: Boolean = false,
        val forMainAdmin: Boolean = false
)